### ISSUE TITLE
Incompatible version of org.apache.tomcat.util.compat.JreCompat

### ISSUE THREAD

**Issue by haymant** [grpc/grpc-web#977]
sprint-boot-starter-web can't start after adding latest grpcweb-java. 

```
An attempt was made to call a method that does not exist. The attempt was made from the following location:

    org.apache.catalina.startup.Tomcat.<clinit>(Tomcat.java:1393)

The following method did not exist:

    org.apache.tomcat.util.compat.JreCompat.isGraalAvailable()Z

The method's class, org.apache.tomcat.util.compat.JreCompat, is available from the following locations:

    jar:file:/home/xxx/.m2/repository/org/mortbay/jasper/apache-jsp/8.5.40/apache-jsp-8.5.40.jar!/org/apache/tomcat/util/compat/JreCompat.class
    jar:file:/home/xxx/.m2/repository/org/apache/tomcat/embed/tomcat-embed-core/9.0.30/tomcat-embed-core-9.0.30.jar!/org/apache/tomcat/util/compat/JreCompat.class

It was loaded from the following location:

    file:/home/xxx/.m2/repository/org/mortbay/jasper/apache-jsp/8.5.40/apache-jsp-8.5.40.jar
```

**Comment by stanley-cheung** [comment]
@vnorigoog Can you please take a look? Thanks.

**Comment by vnorigoog** [comment]
grpcweb-java code at src/connector/src doesn't refer to Tomcat.
What specific code are you referring to when you say "adding latest grpcweb-java".

**Comment by haymant** [comment]
grpcweb-java refers to apache-jsp. Not sure why old 8.5.40 is used: ![image](https://user-images.githubusercontent.com/700512/96799509-d8ea7b80-1435-11eb-8f8e-752c87c08920.png)

apache-jsp calls org.apache.tomcat.util.compat.JreCompat, which conflicts with spring-boot-starter-web is using tomcat's version: 
![image](https://user-images.githubusercontent.com/700512/96800777-b73ec380-1438-11eb-8d16-2e516880ed65.png)

**Comment by vnorigoog** [comment]
oh ok. I see it now.
we don't directly refer to Apache-jsp.
I can try to upgrade jetty-maven-plugin to more recent one (from the Jan/2020 release) but thats no guarantee that it would use the more recent apache-jsp.

The problem here is that spring is referring to a version that is different from what grpcweb-java code indirectly refers to.
I am not sure if there is universal fix for this. Different frameworks can use different versions of the imported plugin's...

**Comment by vnorigoog** [comment]
upgrading to latest jetty plugins in https://github.com/grpc/grpc-web/pull/979

but I can't say it will fix this issue.
anytime we mix multiple libraries importing their own versions of the commonly used apache plugins, this kind of issue will appear unless maven is told to take the latest version of conflicting versions..

**Comment by haymant** [comment]
yes, confirmed it does not work. I am not sure why jetty-maven-plugin refers to apache-jsp 9.4 and at end refers to apache-jsp 8.5.40. Considering popularity of spring boot, it'd be helpful to solve it, even it isn't universal.

**Comment by colltoaction** [comment]
This seems to solve the issue from an end user perspective (i.e I have a Spring Boot + grpc project):

```gradle
configurations {
    runtime.exclude group: 'org.mortbay.jasper', module: 'apache-jsp'
}
```
